config/seed_data.toml

```diff
@@ -6,7 +6,9 @@ last_name = "Testington"
 [[user_roles]]
 user_id = 1
 role = "admin"
+created_at="2025-04-04T16:03:35"
 
 [[user_roles]]
 user_id = 1
 role = "writer"
+created_at="2025-04-04T16:03:35"
```

Comment:
any idea why the `DEFAULT CURRENT TIMESTAMP` on `created_at` doesn't cover this?

Reply:
That's what i was wondering too. New to rust btw but I think that the User_Role struct was expecting created_at to have data. Was wondering if there's a way to make the field optional
https://github.com/foltik/lsd/blob/main/src/db/user.rs#L18-L23

Reply:
ahh of course. it doesn't make sense to make it optional (I'm not sure if SQLx would even allow that) because it is guaranteed to be non-null on the SQL side and the Rust struct should try to exactly replicate the SQL data model.

Reply:
Fixed it with wrapping it with `Option<NaiveDateTime>`

Reply:
I tried it with making it optional and it was able to work. Is there a reason it need to be exactly like the SQL model? is that for consistency or is it important on the Rust side?

Reply:
Mainly consistency in marshalling between the Rust and SQL data models. Putting it in an Option also introduces some development friction in having to check the option whenever it is accessed. I don't think that trade off is worth it just to avoid requiring creation date in the seed files.